playground/frontend/lib/pages/playground/states/examples_state.dart

```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
     return example;
   }
 
-  _loadCategories() async {
-    sdkCategories = await _exampleRepository.getListOfExamples(
-      GetListOfExamplesRequestWrapper(sdk: null, category: null),
-    );
-    await _loadDefaultExamples(sdkCategories);
-    notifyListeners();
+  _loadCategories() {
+    _exampleRepository
+        .getListOfExamples(
+          GetListOfExamplesRequestWrapper(sdk: null, category: null),
+        )
+        .then((map) => setSdkCategories(map));
   }
 
   changeSelectorVisibility() {
     isSelectorOpened = !isSelectorOpened;
     notifyListeners();
   }
 
-  _loadDefaultExamples(sdkCategories) async {
+  loadDefaultExamples() async {
+    if (defaultExamplesMap != null) {
+      return;
+    }
     defaultExamplesMap = {};
-    List<MapEntry<SDK, ExampleModel>> entries = [];
-    for (SDK sdk in SDK.values) {
-      ExampleModel? defaultExample = sdkCategories![sdk]?.first.examples.first;
-      if (defaultExample != null) {
-        // load source and output async
-        loadExampleInfo(defaultExample, sdk);
-        entries.add(MapEntry(sdk, defaultExample));
-      }
+    List<MapEntry<SDK, ExampleModel>> defaultExamples = [];
+
+    for (var value in SDK.values) {
+      defaultExamples.add(
+        MapEntry(
+            value,
+            await _exampleRepository.getDefaultExample(
+              GetExampleRequestWrapper('', value),
+            )),
+      );
+    }
+
+    defaultExamplesMap!.addEntries(defaultExamples);
```

Comment:
Is this exclamation mark is needed? The object is initialized on line 108.

Reply:
I'll remove it, thank you!